### DISCUSSION TITLE
5.x support in plugins

### DISCUSSION THREAD

**Discussion by AndrolGenhald** [vimeo/psalm#8291] [Psalm development (AKA internals)]
@orklah @weirdan

How do we want to approach 5.x support in the official plugins?

New major version of the plugin, or is minor version ok?

Support only 5.x going forward or either:
 - Support both by having a requirement of `^4.0|^5.0`
 - Support both by having separate branches/major versions and backport changes

Whether or not we drop support for 4.x in plugins after 5.x is released, we'll want to keep supporting it until then.

**Comment by orklah** [comment]
I don't have enough hindsight on what each choice would entail.

What I can say is that we don't have the manpower to support a lot of unnecessary things so I'd say we should choose the simplest path, even if there are downsides for users.

For example, if we want to migrate a plugin, I'd rather make a new version that only support Psalm 5 instead of having to build compatibility layers.
